src/package.ts

```diff
@@ -0,0 +1,92 @@
+import { privateEncrypt } from 'crypto';
+import contextKeys from './contextKeys';
+import { exec, pathExists } from './utilities';
+
+// Swift Package Manager contents
+export interface PackageContents {
+    name: string
+    products: Product[]
+    dependencies: Dependency[]
+    targets: Target[]
+}
+
+// Swift Package Manager product
+export interface Product {
+    name: string
+    targets: string[]
+    type: 'library'|'executable'
+}
+
+// Swift Package Manager target
+export interface Target {
+    name: string
+    path: string
+    sources: string[]
+    type: 'executable'|'test'|'library'
+}
+
+// Swift Package Manager dependency
+export interface Dependency {
+    identity: string
+    url?: string
+    path?: string
+}
+
+// Class holding Swift Package Manager Package
+export class SwiftPackage implements PackageContents {
```

Comment:
Just a general comment concerning filenames:

TypeScript doesn't seem to have any strict rules, so the convention I've adopted was: If a file's primary export is a type or object, name it after that type or object (in UpperCamelCase and lowerCamelCase respectively). Otherwise, pick an appropriate lowerCamelCase name.

By that convention, we could name the new files `SwiftPackage` and `SwiftContext`. Any thoughts on this convention?

CC @0xTim

Reply:
I'm not fussed either way. Given the symbol look up is pretty solid I don't think it is a huge issue. But I'm not against renaming either